DigitalLearningSolutions.Web/Controllers/TrackingSystem/CentreConfiguration/RegistrationPromptsController.cs

```diff
@@ -55,7 +66,15 @@ public IActionResult EditRegistrationPrompt(int promptNumber)
             var customPrompt = customPromptsService.GetCustomPromptsForCentreByCentreId(centreId).CustomPrompts
                 .Single(cp => cp.CustomPromptNumber == promptNumber);
 
-            return View(new EditRegistrationPromptViewModel(customPrompt));
+            var data = TempData.Peek<EditRegistrationPromptData>();
+
+            var model = data != null

```

Comment:
We can use TempData Get instead of Peek here as it clears the tempdata by default. :)

Reply:
There is no such strongly typed method for this. This will result in something like this:
`var model = TempData[nameof(EditRegistrationPromptData)] is EditRegistrationPromptData data
                ? data.EditModel!
                : new EditRegistrationPromptViewModel(customPrompt);`
Or should I try and write one in a similar pattern to the Peek<T> method?

Reply:
Nevermind, that just completely doesn't work. The options are either keep the existing Peek/Clear flow or write a new extension. Will try to write the extension as it'll probably be useful later.

Reply:
Yes, a new extension sounds great! 👍